### PR TITLE
Add the official Python 3.10 support

### LINKED ISSUE #476: Build failure in python 3.10. Any plan supporting 3.10 soon?
Hi all,
We have an indirect dependency to py4j through another package and the latest 0.10.9.4 release is failing with the following error:
```
from py4j.clientserver import ClientServer
/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/site-packages/py4j/clientserver.py:12: in <module>
    from collections import deque, Callable
E   ImportError: cannot import name 'Callable' from 'collections' (/opt/hostedtoolcache/Python/3.10.2/x64/lib/python3.10/collections/__init__.py)
```

The build was working fine under 0.10.9.3 in python 3.10. Any plan to support python 3.10 soon?

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR adds the official support of Python 3.10:
- Fix CI to test Python 3.10.
- Fix the documentation to note Python 3.10 support.

Resolves #476

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Gonna merge this and roll another release to minimize the impact of the previous release (that broke the unofficia Python 3.10 accidentally)

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       matrix:
         os: [ ubuntu-18.04 ]
         java-version: [ 8 ]
-        python-version: [ 2.7, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9 ]
+        python-version: [ '2.7', '3.4', '3.5', '3.6', '3.7', '3.8', '3.9', '3.10' ]
```
Comment (dongjoon-hyun) [inline .github/workflows/test.yml:18]: Thank you for adding 3.10.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -56,18 +56,17 @@ jobs:
           ./gradlew check
           ./gradlew assemble
 
-      # There is no combination of flake8 and pyflakes versions which
-      # are compatible with the entire range of Python versions tested
-      # in our build matrix. Therefore, only run flake8 on the latest
-      # Python version:
-      - name: Run flake8
-        if: ${{ matrix.python-version == '3.9' }}
+      # Nose is no logner maintained, and does not support Python 3.10.
```
Comment (bartdag) [inline .github/workflows/test.yml:59]: > # Nose is no logner maintained
no longer

We could use pytest for all versions, but we will need an old release for 2.7 and 3.4 (4.6.9 I believe).

**Review by bartdag — COMMENTED** [review]
Thank you for this addition!

**Comment by sayanchk** [comment]
Thanks for adding python 3.10 support!

**Pull request merged**